storybook/manager-head.html

```diff
@@ -0,0 +1,35 @@
+<script>
+	( function redirectIfStoryMoved() {
+		const REDIRECTS = [
+			{
+				from: /\/components-deprecated-/,
+				to: '/components-',
+			},
+		];
+
+		const params = new URLSearchParams( window.location.search );
+
+		const matchedRedirect = REDIRECTS.find( ( { from } ) =>
+			from.test( params.get( 'path' ) )
+		);
+
+		if ( ! matchedRedirect ) {
+			return;
+		}
+
+		params.set(
+			'path',
+			params
+				.get( 'path' )
+				.replace( matchedRedirect.from, matchedRedirect.to )
+		);
+
+		const newUrl =
+			window.location.origin +
+			window.location.pathname +
+			'?' +
+			decodeURIComponent( params.toString() );
```

Comment:
Should this just use the `URL` constructor just in case? Something like:

```
new URL( window.location.pathname, window.location.origin )
```

Also, `params` being a `URLSearchParams` will serialize to a string anyway:

```
'' + new URLSearchParams( { x: 1, y: 'z' } ) // 'x=1&y=z'
```

so this could just be:

```
const { pathname, origin } = window.location;
const newUrl = new URL( pathname, origin ) + '?' + params;
```

Reply:
Good call 👍